sql/src/main/java/io/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,435 @@
+    {
+      return SERVERSEGMENTS_SIGNATURE.getRelDataType(typeFactory);
+    }
+
+    @Override
+    public TableType getJdbcTableType()
+    {
+      return TableType.SYSTEM_TABLE;
+    }
+
+    @Override
+    public Enumerable<Object[]> scan(DataContext root)
+    {
+      final List<Object[]> rows = new ArrayList<>();
+      final Map<String, QueryableDruidServer> serverViewClients = serverView.getClients();
+      for (QueryableDruidServer queryableDruidServer : serverViewClients.values()) {
+        Object[] row = new Object[SERVERSEGMENTS_TABLE_SIZE];
+        final DruidServer druidServer = queryableDruidServer.getServer();
+        final Map<String, DataSegment> segmentMap = druidServer.getSegments();
+        for (DataSegment segment : segmentMap.values()) {
+          row[0] = druidServer.getHost();
+          row[1] = segment.getIdentifier();
+          rows.add(row);
+        }
+      }
+      return Linq4j.asEnumerable(rows);
+    }
+  }
+
+  private static class TasksTable extends AbstractTable implements ScannableTable
+  {
+    private final DruidLeaderClient druidLeaderClient;
+    private final ObjectMapper jsonMapper;
+
+    public TasksTable(
+        DruidLeaderClient druidLeaderClient,
+        ObjectMapper jsonMapper
+    )
+    {
+      this.druidLeaderClient = druidLeaderClient;
+      this.jsonMapper = jsonMapper;
+    }
+
+    @Override
+    public RelDataType getRowType(RelDataTypeFactory typeFactory)
+    {
+      return TASKS_SIGNATURE.getRelDataType(typeFactory);
+    }
+
+    @Override
+    public TableType getJdbcTableType()
+    {
+      return TableType.SYSTEM_TABLE;
+    }
+
+    @Override
+    public Enumerable<Object[]> scan(DataContext root)
+    {
+      final List<TaskStatusPlus> tasks = getTasks(druidLeaderClient, jsonMapper);
+      final FluentIterable<Object[]> results = FluentIterable
+          .from(tasks)
+          .transform(
+              task -> new Object[]{
+                  task.getId(),
+                  task.getType(),
+                  task.getDataSource(),
+                  task.getCreatedTime(),
+                  task.getQueueInsertionTime(),
+                  task.getState(),
+                  task.getRunnerTaskState(),
+                  task.getDuration(),
+                  task.getLocation() != null
+                    ? task.getLocation().getHost() + ":" + (task.getLocation().getTlsPort()
+                                                          == -1
+                                                          ? task.getLocation()
+                                                                .getPort()
+                                                          : task.getLocation().getTlsPort())
+                    : null,
+                  task.getErrorMsg()
+              }
+          );
+
+      return Linq4j.asEnumerable(results);
+    }
+
+    private List<TaskStatusPlus> getTasks(
+        DruidLeaderClient druidLeaderClient,
```

Comment:
Let's name this what it is (indexing service client).

Reply:
done